From a356d83306605f19b206ec13dc71609deae7cbfd Mon Sep 17 00:00:00 2001
From: Wen He <wen.he_1@nxp.com>
Date: Wed, 11 Sep 2019 14:11:07 +0800
Subject: [PATCH 109/245] drm/imx/hdp: Correction the core clock of the DP
 Controller

commit 93e838b3e1df82d9d76a7e87066631286bf4cbe6 from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux.git

The wrapper instantiating both the LCD Controller and the DP
Controller forecs this to be a synchronous integer division of
the LCD Controller's AXI(ACLK) interface, so this is connected
to /4 of the ACLK frequency.

Signed-off-by: Wen He <wen.he_1@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/gpu/drm/imx/hdp/imx-dp.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/gpu/drm/imx/hdp/imx-dp.c b/drivers/gpu/drm/imx/hdp/imx-dp.c
index 2864e96dd7b4..3be42a6e3370 100644
--- a/drivers/gpu/drm/imx/hdp/imx-dp.c
+++ b/drivers/gpu/drm/imx/hdp/imx-dp.c
@@ -40,6 +40,9 @@ int hdp_fw_init(state_struct *state)
 
 	core_rate = clk_get_rate(hdp->clks.clk_core);
 
+	/* this clock will be ACLK/4 */
+	core_rate = core_rate/4;
+
 	/* configure the clock */
 	CDN_API_SetClock(state, core_rate/1000000);
 
-- 
2.17.1

